docs/key-derivation-design.md

```diff
@@ -0,0 +1,80 @@
+
+### Approach
+
+Since `t` valid shares are required to authorise a coordinator there is a natural credential that a coordinator can present to demonstrate they've talked to devices with `t` valid shares: The root polynomial image.
+This is the polynomial the coordinator uses to verify signature shares under that access structure and can be reconstructed from `t` valid share images.
+
+Therefore we make each device present its share images when it connects to a coordinator. Once a coordinator has `t` of them it can reconstruct the root polynomial image and from then on it can present some kind of credential derived from it to authorise itself as a coordinator to request signatures under that access structure.
+
+It's important to understand that we are not making requesting signatures more restrictive than it needs to be. One could ask, "Why not just let any old coordinator request signatures? If they don't have access to enough devices they won't be able to finish the signature anyway".
+The problem with this (very natural) approach is the coordinator would still need to be told the root public key so it can create and verify the signing session anyway.
+Having a single device provide the coordinator all this information at this stage would mean in the context of Bitcoin, you would have revealed the wallet's XPUB to anyone who happens upon a single frostsnap device (without some very awkward designs to keep the XPUB secret).
+Furthermore it's unlikely they could start a signing session yet since the coordinator presumably doesn't have nonces from the other devices yet. If they had nonces they could just as easily have acquired share images from the other devices too so we're back at our original design.
+
+Since the root polynomial is a universal credential that only authenticated coordinators have and is independent of any specific coordinator we can also use it to encrypt shares on each device.
+Devices can forget about the root polynomial after key generation but before they do, encrypt their new key share with a key derived from it.
+
+Now an attacker who finds a device learns nothing from it other than a share image for each share the device has. This tells them only how many keys that devices was involved with and their threshold `t` (and by the index a little bit more info about `n`). The attacker can only start requesting signatures after they discover `t` devices which satisfies our security assumption.
+
+### Concrete key derivations
+
+Here's how keys are derived. Remember `root_key` is the first coefficient of the root polynomial generated during key generation.
+
+```
+root_key (m) // root_key includes a chain code of 32 zeros
+└── app_key (m/0)
+    ├── TestMessage (m/0/0)
+    | // Descriptors/PSBT will treat this as the root for everything below
+    ├── Bitcoin (m/0/1)
```

Comment:
One thought is that we could have Bitcoin at `m/0/0`, might be worth keeping the bitcoin derivation as the simplest one. Could get opinions from the matrix channel.

Reply:
I changed Bitcoin to derivation index 0.